predicators/approaches/nsrt_learning_approach.py

```diff
@@ -74,6 +74,27 @@ def _learn_nsrts(self, trajectories: List[LowLevelTrajectory],
             ground_atom_dataset = utils.create_ground_atom_dataset(
                 trajectories, self._get_current_predicates())
             utils.save_ground_atom_dataset(ground_atom_dataset, dataset_fname)
+        elif CFG.offline_data_method in ["demo+labelled_atoms", "img_demos"]:
+            # In this case, the annotations aren't basically ground atoms!
+            # We can use these to make GroundAtomTrajectories.
+            assert annotations is not None
+            assert len(annotations) == len(trajectories)
+            ground_atom_dataset = []
+            annotations_with_only_known_preds = []
```

Comment:
I'm trying to understand the name/meaning of `annotations_with_only_known_preds`. I'm a bit generally confused. In the VLM setting, are there any annotations that aren't known predicates? 

I understand that: we have some given predicates (goal predicates), and then the VLM comes up with some other predicates, and we subselect out of those after grammar search invention approach is done, to get our final set of predicates. 
^ in this context, what is a known predicate?

Reply:
will rename!